src/Node/Expression/Binary/ElvisBinary.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Node\Expression\Binary;
+
+use Twig\Compiler;
+use Twig\Node\Expression\OperatorEscapeInterface;
+
+final class ElvisBinary extends AbstractBinary implements OperatorEscapeInterface
+{
+    public function operator(Compiler $compiler): Compiler
+    {
+        return $compiler->raw('?:');
+    }
+
+    public function getOperandNamesToEscape(): array
+    {
+        return ['left', 'right'];
```

Comment:
This implementation looks wrong to me. When applying auto-escaping, we need to escape the `then` part without escaping the condition itself (to avoid changing the condition), which means turning `a ?: b` into `a ? a|e : b|e`

Reply:
Right, fixed.